tests/test_using_zigzag_inside_pytest.py

```diff
@@ -0,0 +1,125 @@
+    testdir.makepyfile(single_decorated_test_function.format(mark_type=mark_type_exp,
+                                                             mark_arg=test_id_exp,
+                                                             test_name=test_name_exp))
+    result_path = testdir.tmpdir.join('junit.xml')
+
+    # mock
+    mocker.patch('zigzag.zigzag.ZigZag.__init__', return_value=None)
+    mocker.patch('zigzag.zigzag.ZigZag.upload_test_results')
+
+    result = testdir.runpytest(
+        "--junitxml={}".format(result_path),
+        "--ci-environment=mk8s",
+        "--zigzag",
+        "--qtest-project-id={}".format(project_id))
+
+    # Test
+    assert 'ZigZag upload was successful!' not in result.outlines
+    assert 'The ZigZag upload was not successful' in result.outlines
+    assert "'QTEST_API_TOKEN'" in result.outlines
+
+
+def test_zigzag_no_project_id(testdir, single_decorated_test_function, mocker):
+    """Verify that zigzag will tell the user to supply a project-id"""
+
+    # Setup
+    mark_type_exp = 'test_id'
+    test_id_exp = '123e4567-e89b-12d3-a456-426655440000'
+    test_name_exp = 'test_uuid'
+    env_vars = {'QTEST_API_TOKEN': 'valid_token'}
+    testdir.makepyfile(single_decorated_test_function.format(mark_type=mark_type_exp,
+                                                             mark_arg=test_id_exp,
+                                                             test_name=test_name_exp))
+    result_path = testdir.tmpdir.join('junit.xml')
+
+    # mock
+    mocker.patch('zigzag.zigzag.ZigZag.__init__', return_value=None)
+    mocker.patch('zigzag.zigzag.ZigZag.upload_test_results')
+    mocker.patch.dict('os.environ', env_vars)
+
+    result = testdir.runpytest(
+        "--junitxml={}".format(result_path),
+        "--ci-environment=mk8s",
+        "--zigzag")
+
+    # Test
+    assert 'The ZigZag upload was not successful' in result.outlines
+    assert "'qtest-project-id' is required when using ZigZag" in result.outlines
+
+
+def test_no_zigzag(testdir, single_decorated_test_function, mocker):
+    """Verify zigzag does not run if we do no pass the --zigzag option"""
+
+    # Setup
+    mark_type_exp = 'test_id'
+    test_id_exp = '123e4567-e89b-12d3-a456-426655440000'
+    test_name_exp = 'test_uuid'
+    project_id = '12345'
+    env_vars = {'QTEST_API_TOKEN': 'valid_token'}
+    testdir.makepyfile(single_decorated_test_function.format(mark_type=mark_type_exp,
+                                                             mark_arg=test_id_exp,
+                                                             test_name=test_name_exp))
+    result_path = testdir.tmpdir.join('junit.xml')
+
+    # mock
+    mocker.patch.dict('os.environ', env_vars)
+
+    result = testdir.runpytest(
+        "--junitxml={}".format(result_path),
+        "--ci-environment=mk8s",
+        "--qtest-project-id={}".format(project_id))
+
+    # Test
+    for line in result.outlines:
+        assert 'ZigZag' not in line
```

Comment:
Should we place a warning in the terminal report about the fact that `--qtest-project-id` was specified without the `--zigzag` option?

Reply:
No thoughts on this one?

Reply:
I feel like this in an improvement that we can make once we get the river flowing the other direction.

Reply:
@zreichert Create a ticket and I'll approve.